spyder/app/mainwindow.py

```diff
@@ -2969,10 +2972,23 @@ def open_external_file(self, fname):
         variable explorer inside Spyder.
         """
         fname = encoding.to_unicode_from_fs(fname)
-        if osp.isfile(fname):
-            self.open_file(fname, external=True)
-        elif osp.isfile(osp.join(CWD, fname)):
-            self.open_file(osp.join(CWD, fname), external=True)
+        if osp.exists(osp.join(CWD, fname)):
+            fpath = osp.join(CWD, fname)
+        elif osp.exists(fname):
+            fpath = fname
+        else:
+            return
+
+        if osp.isfile(fpath):
+            self.open_file(fpath, external=True)
+        elif osp.isdir(fpath):
+            QMessageBox.warning(
+                None, "Spyder",

```

Comment:
Please @akdor1154 lets change `None` for `self` and set as title `Warning` instead of `Spyder`.

Reply:
Let's use `_("Error")` instead of `Warning` or `Spyder` for this dialog's title, because this is really an error.